src/server/search/search_family.cc

```diff
@@ -398,20 +398,40 @@ void SearchFamily::FtInfo(CmdArgList args, ConnectionContext* cntx) {
   for (const auto& info : infos)
     total_num_docs += info.num_docs;
 
-  const auto& schema = infos.front().base_index.schema;
+  const auto& info = infos.front();
+  const auto& schema = info.base_index.schema;
 
-  (*cntx)->StartCollection(3, RedisReplyBuilder::MAP);
+  (*cntx)->StartCollection(4, RedisReplyBuilder::MAP);
 
   (*cntx)->SendSimpleString("index_name");
   (*cntx)->SendSimpleString(idx_name);
 
-  (*cntx)->SendSimpleString("fields");
+  (*cntx)->SendSimpleString("index_definition");
+  {
+    (*cntx)->StartCollection(2, RedisReplyBuilder::MAP);
+    (*cntx)->SendSimpleString("key_type");
+    (*cntx)->SendSimpleString(info.base_index.type == DocIndex::JSON ? "JSON" : "HASH");
+    (*cntx)->SendSimpleString("prefix");
+    (*cntx)->SendSimpleString(info.base_index.prefix);
+  }
+
+  (*cntx)->SendSimpleString("attributes");
   (*cntx)->StartArray(schema.fields.size());
   for (const auto& [field_ident, field_info] : schema.fields) {
-    string_view reply[6] = {"identifier", string_view{field_ident},
-                            "attribute",  field_info.short_name,
-                            "type"sv,     SearchFieldTypeToString(field_info.type)};
-    (*cntx)->SendSimpleStrArr(reply);
+    vector<string> info;
+
+    string_view base[] = {"identifier"sv, string_view{field_ident},
+                          "attribute",    field_info.short_name,
+                          "type"sv,       SearchFieldTypeToString(field_info.type)};
+    info.insert(info.end(), base, base + ABSL_ARRAYSIZE(base));
+
+    if (field_info.flags & search::SchemaField::NOINDEX)
+      info.push_back("NOINDEX");
+
+    if (field_info.flags & search::SchemaField::SORTABLE)
+      info.push_back("SORTABLE");
```

Comment:
yes, this format is really this convoluted 😵 ... The first parts are k-v pairs, afterwards just the flags & details follow. The first few keys are all lowercase, all other stuff is uppercase